analytics2/utils/block_stats.py

```diff
@@ -198,6 +199,73 @@ def add_block_pop_density(bldg_pop: gpd.GeoDataFrame,
     return bldg_pop  
 
 
+def add_block_gini(bldg_pop: gpd.GeoDataFrame,
+                   values_col: str,
+                   block: Union[gpd.GeoDataFrame, str] = None,
+                   weights: libpysal.weights.weights.W = None
```

Comment:
the type of `block` is `Optional[Union[gpd.GeoDataFrame, str]]`, similar thing for `weights`